src/python/pants/core/util_rules/strip_source_roots.py

```diff
@@ -30,6 +33,27 @@ class SourceRootStrippedSources:
     """Wrapper for a snapshot of files whose source roots have been stripped."""
 
     snapshot: Snapshot
+    # source root -> file paths relative to that root.
+    # Note that this will not contain entries for files to which the concept of source roots
+    # doesn't apply (e.g., the sources of a files(...) target), even when such files are
+    # in the snapshot.
+    root_to_relfiles: FrozenDict[str, Tuple[str, ...]]
+
+    @classmethod
+    def for_single_source_root(
+        cls, snapshot: Snapshot, source_root: str
```

Comment:
```suggestion
        cls, stripped_snapshot: Snapshot, source_root: str
```

Consider also requiring keyword args. I could see people using this improperly by putting the original unstripped snapshot in the first arg. A keyword arg would make sure people didn't do that.

Reply:
Eh, this matches the pre-existing field name, so I'm inclined to leave it as-is for consistency.